[Kingston] add flow for reporting bins not returned or contents spilled

struan opened the pull request:
This is mostly based on the [Sutton code](https://github.com/mysociety/fixmystreet/commit/0cc3311cf5) with a few Kingston specific changes to add checkbox extra details to the bin not returned, plus an extra message for FAS properties for spillages.   

Fixes mysociety/societyworks#5245
Fixes mysociety/societyworks#5246

[skip changelog]

struan committed:
[Kingston] Bins not returned/waste spillage.

This is mostly based on the Sutton code from 0cc3311cf5 with a few
Kingston specific changes to add checkbox extra details to the bin not
returned, plus a message for FAS properties for spillages.

struan committed:
[Kingston] Bins not returned/waste spillage.

This is mostly based on the Sutton code from 0cc3311cf5 with a few
Kingston specific changes to add checkbox extra details to the bin not
returned, plus a message for FAS properties for spillages.

dracos commented on templates/email/default/waste/other-reported.txt:
```diff
@@ -55,6 +55,14 @@ On delivery day - You do not need to be home for our crew to deliver the bin.
 Collecting your old bin - If you've requested a collection of your old bin, we'll schedule a specific date for our
 crew to remove it. We aim for this to be the same day as we deliver your new bin, but it could be a different day.
 
+[% ELSIF cobrand.moniker == 'kingston' AND report.category == 'Waste spillage' %]
```
Also could factor it together with the Sutton ones in this file as they're identical - but I guess might change 🤷

dracos reviewed (changes requested):
Looks good, some refactoring mostly I think :)

dracos commented on templates/email/default/waste/other-reported.txt:
```diff
@@ -48,7 +48,8 @@ We apologise for any inconvenience this may have caused and appreciate your feed
 [% END %]
 
-[% ELSIF cobrand.moniker == 'kingston' AND report.category == 'Request new container' %]
+[% ELSIF cobrand.moniker == 'kingston'%]
```
There's one more kingston check unneeded in the second sub-IF.

dracos commented on templates/web/kingston/waste/enquiry-problem.html:
```diff
@@ -13,10 +13,7 @@
 END;
 
 IF service.report_allowed AND property.domestic_refuse_bin;
-    SET field = {
```
If this was meant to make the file a symlink, I don't think it's worked, dunno if there's some git thing going on here.

dracos commented on templates/web/kingston/waste/enquiry-problem.html:
```diff
@@ -0,0 +1,29 @@
+[%
```
(Don't think this has worked)

dracos reviewed (changes requested):
Tiny issue with the email template to resolve, symlink thing I'm sure we can fix.

dracos reviewed (approved):
Woo; still got that superfluous `ELSIF cobrand.moniker == 'kingston'` in the text email but doesn't matter if it's left in :)

struan committed:
[Sutton/Kingston] Move escalations to shared Role.

Now that Kingston will be using escalations move the escalation code out
of the Sutton cobrand and into the shared Role (but with Kingston not
running it at present); also sets up for other future use.

struan committed:
[Waste][Echo] add all_events details to Kingston as well as Sutton

This is used by the escalation code to check if there's an existing
event that could be escalated

struan committed:
[Waste] fix incorrect object name for bulky comment list in html emails

dracos committed:
[SLWP] Update bulky resolution codes.

dracos commented on templates/web/sutton/waste/enquiry-problem.html:
```diff
@@ -94,7 +110,10 @@
     SET field = { value = 'Waste spillage', label = 'Report a spillage by the crew' };
-    IF service.report_allowed AND property.domestic_refuse_bin;
+    IF c.cobrand.moniker == 'kingston' AND fas_property;
+        field.hint = 'Follow the link in the introduction to request a street clean';
```
Not sure I'd bother with the hint.
(It's possible from the KGS272 internal comparison doc that they're happy for FAS to continue here, unclear if that changes the previous request...)

dracos commented on templates/web/sutton/waste/enquiry-problem.html:
```diff
@@ -94,7 +110,10 @@
+    IF c.cobrand.moniker == 'kingston' AND fas_property;
+        field.hint = 'Follow the link in the introduction to request a street clean';
+        field.disabled = 1;
+    ELSIF service.report_allowed AND property.domestic_refuse_bin;
```
I don't think the IF domestic should be here, sorry, missed that in previous PR, otherwise this just means it isn't checking for 'already reported a waste spillage' for communal/FAS properties.

dracos commented on templates/web/sutton/waste/enquiry-problem.html:
```diff
@@ -33,11 +33,27 @@
         on [% date.format(service.last.date) | replace('~~~', service.last.ordinal) %].
         If it has not been collected, you can report it as missed and we will return within 2 working days to collect it.
         </p>
+    [% ELSIF c.cobrand.moniker == 'kingston' AND (property.domestic_refuse_bin OR communal_property OR fas_property)%]
```
Few things on this section:
a) Think "property.domestic_refuse_bin OR communal_property OR fas_property" covers all properties? :)
b) Link for Kingston has a typo in 'street'
c) I don't think all this text should be here. It can be shown after they pick the category, as an extra data notice on the category (which is what currently happens on live, see the live Sutton "Waste spillage" category for example), and this can then be simplified and not repeat the date part.
d) The text here isn't what's given in the final document https://docs.google.com/document/d/193SMUDL-JOlm8qi0_XXgsdHO_OWBgQ0E58naNsw8cWE/edit?tab=t.0 under "Spillage by Crew" - so that text would be what's then added as an extra data notice on each category.
e) Then as per above doc, once that's done, I think https://github.com/mysociety/societyworks/issues/5575 then adds on nicely, in that the template needs updating to stay the same for Sutton but to always show the Bin not returned/Spillage options for Kingston.

So this bit I'd have something like
```
    [% ELSIF service.report_allowed %]
        <p>Our records show this collection took place
        on [% date.format(service.last.date) | replace('~~~', service.last.ordinal) %].
        If it has not been collected, you can report it as missed and we will return within 2 working days to collect it.
        </p>
        [% IF cobrand.moniker == 'kingston' AND fas_property %]
            <p>If a spillage on a public highway occurred during a [% unit.service_name FILTER lower %] collection, please report <a href="https://kingston-self.achieveservice.com/service/Request_a_street_clean_in_Kingston">a street cleaning issue</a>.</p>
       [% END %]
```
